### ISSUE TITLE
Add trackio as an experiment tracker alongside wandb

### ISSUE THREAD

**Issue by zuoxingdong** [huggingface/lerobot#4319]
### Ticket Type

💡 Feature Request / Improvement

### Environment & System Info

```Shell

```

### Description

**Motivation:** let `lerobot-train` log to [trackio](https://github.com/gradio-app/trackio) as an alternative to wandb.

Now `wandb` is the only way to get a training curve out of `lerobot-train`. That means an account
and an API key before you can see whether your loss is going down, which is friction for a first
training run and a hard stop on machines that cannot reach wandb. 

[Trackio](https://github.com/gradio-app/trackio) needs neither: runs go to a local SQLite database you browse with `trackio show`, and if you do want to look at a run from elsewhere, the dashboard can be hosted on a Hugging Face Space.

The tracking config is currently wandb-specific (`cfg.wandb`, `--wandb.enable=true`,
`WandBLogger` constructed inline in `lerobot_train.py` and `rl/learner.py`), so a second tracker
cannot be added without either special-casing it in the training scripts or reusing fields named
after wandb.

**Proposal:** introduce a tracker choice registry, so trackers are selected the way policies,
optimizers and environments already are:

```bash
lerobot-train ... --tracker.type=wandb    # what --wandb.enable=true does today
lerobot-train ... --tracker.type=trackio  # new
```

with a small `TrackerLogger` protocol (`log_dict` / `log_policy` / `log_video`) and a
`make_tracker(cfg)` factory, so the training scripts stop knowing which backend they are talking to
and a third tracker later is a config subclass plus a logger.

Existing `--wandb.*` flags would keep working unchanged and simply build the equivalent wandb
tracker, so no existing command, config or checkpoint breaks.

**PR**: #4320 

### Context & Reproduction

**End to end in about a minute, no account and no API key** 

```bash
pip install -e ".[trackio]"

lerobot-train \
    --dataset.repo_id=lerobot/pusht \
    --policy.type=act \
    --policy.push_to_hub=false \
    --steps=200 \
    --save_checkpoint=false \
    --tracker.type=trackio \
    --tracker.project=trackio-demo

trackio show --project trackio-demo
```

Design points that came up while implementing this, and where maintainer input would help most:

1. **Should the `--wandb.*` flags be deprecated?** My patch keeps them working and silently builds
   the equivalent tracker, with no deprecation warning, because deprecating a documented flag feels
   like a call for the team rather than something to settle inside a feature PR. Happy to add the
   warning and update the remaining docs if you would rather move users over.
2. **`WandBConfig.enable` has no meaning once a tracker is selected**, since selecting one is what
   enables it (and `--tracker=null` turns tracking off again). The wandb tracker config inherits the
   field from the existing `WandBConfig` and I neutralize it rather than duplicating the field list;
   there is a cleaner shape that costs one extra dataclass, and I am happy to go either way.

### Relevant logs or stack trace

```Shell

```

### Checklist

- [x] I have searched existing tickets to ensure this isn't a duplicate.
- [x] I am using the latest version of the `main` branch.
- [x] I have verified this is not an environment-specific problem.
